catroid/src/main/java/org/catrobat/catroid/ProjectManager.java

```diff
@@ -793,6 +798,31 @@ private boolean checkReferencesOfCurrentBrick(Brick currentBrick) {
 				return false;
 			}
 		}
+
+		if (currentBrick instanceof IfThenLogicBeginSimpleBrick) {
+			IfThenLogicEndSimpleBrick endBrick = ((IfThenLogicBeginSimpleBrick) currentBrick)
+					.getIfThenEndSimpleBrick();
+			if (endBrick == null || endBrick.getIfBeginSimpleBrick() == null
+					|| !endBrick.getIfBeginSimpleBrick().equals(currentBrick)) {
+				Log.d(TAG, "Brick has wrong reference:" + currentSprite + " "
+						+ currentBrick);
+				return false;
+			}
+		} else if (currentBrick instanceof IfLogicBeginSimpleBrick) {
+			IfLogicElseSimpleBrick elseBrick = ((IfLogicBeginSimpleBrick) currentBrick).getIfElseSimpleBrick();
+			IfLogicEndSimpleBrick endBrick = ((IfLogicBeginSimpleBrick) currentBrick).getIfEndSimpleBrick();
+			if (elseBrick == null || endBrick == null || elseBrick.getIfEndSimpleBrick() == null
```

Comment:
I know that this is the way it was done before, but this is really hard to read.
At least put the whole simple bricks block and the "normal" if then else block in separate methods